bookkeeper-server/src/test/java/org/apache/bookkeeper/suites/BookKeeperClusterTestSuite.java

```diff
@@ -0,0 +1,146 @@
+        return newServerConfiguration(port, f, new File[] { f });
+    }
+
+    protected static ServerConfiguration newServerConfiguration(int port, File journalDir, File[] ledgerDirs) {
+        ServerConfiguration conf = new ServerConfiguration(baseServerConf);
+        conf.setBookiePort(port);
+        conf.setJournalDirName(journalDir.getPath());
+        String[] ledgerDirNames = new String[ledgerDirs.length];
+        for (int i = 0; i < ledgerDirs.length; i++) {
+            ledgerDirNames[i] = ledgerDirs[i].getPath();
+        }
+        conf.setLedgerDirNames(ledgerDirNames);
+        conf.setEnableTaskExecutionStats(true);
+        return conf;
+    }
+
+    @BeforeClass
+    public static void setUpCluster() throws Exception {
+        setUpCluster(NUM_BOOKIES);
+    }
+
+    protected static void setUpCluster(int numBookies) throws Exception {
+        // set up the metadata store
+        metadataStore = new ZKMetadataStore();
+        metadataStore.start();
+        ServiceURI uri = metadataStore.getServiceUri();
+        log.info("Setting up cluster at service uri : {}", uri.getUri());
+
+        baseClientConf = new ClientConfiguration()
+            .setMetadataServiceUri(uri.getUri().toString());
+        baseServerConf = TestBKConfiguration.newServerConfiguration()
+            .setMetadataServiceUri(uri.getUri().toString());
+
+        // format the cluster
+        assertTrue(BookKeeperAdmin.format(baseServerConf, false, true));
+
+        // start bookies
+        startNumBookies(numBookies);
+    }
+
+    private static void startNumBookies(int numBookies) throws Exception {
+        for (int i = 0; i < numBookies; i++) {
+            ServerConfiguration conf = newServerConfiguration();
+            log.info("Starting new bookie on port : {}", conf.getBookiePort());
+            BookieServer server = startBookie(conf);
+            synchronized (BOOKIES) {
+                BOOKIES.add(server);
+            }
+        }
+    }
+
+    private static BookieServer startBookie(ServerConfiguration conf) throws Exception {
+        conf.setAutoRecoveryDaemonEnabled(true);
+        TestStatsProvider provider = new TestStatsProvider();
+        BookieServer server = new BookieServer(conf, provider.getStatsLogger(""));
+        server.start();
+        return server;
+    }
+
+    @AfterClass
+    public static void tearDownCluster() throws Exception {
+        // stop bookies
+        stopBookies();
+
+        // stop metadata store
+        metadataStore.close();
+        log.info("Stopped the metadata store.");
+
+        // clean up temp dirs
+        for (File f : TMP_DIRS) {
+            FileUtils.deleteDirectory(f);
+        }
+        log.info("Clean up all the temp directories.");
+    }
+
+    private static void stopBookies() {
+        synchronized (BOOKIES) {
+            BOOKIES.forEach(BookieServer::shutdown);
+            log.info("Stopped all the bookies.");
+        }
+    }
+
```

Comment:
in general I would suggest you to consider moving common code to some utility classes and reuse them in BookKeeperClusterTestCase and BookKeeperClusterTestSuite.